keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/NotificationEvent.kt

```diff
@@ -1,13 +1,39 @@
 package com.netflix.spinnaker.keel.events
 
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.core.api.EnvironmentArtifactPin
+import com.netflix.spinnaker.keel.core.api.PinnedEnvironment
 import com.netflix.spinnaker.keel.notifications.Notification
 import com.netflix.spinnaker.keel.notifications.NotificationScope
 import com.netflix.spinnaker.keel.notifications.NotificationType
 
-data class NotificationEvent(
-  val scope: NotificationScope,
-  val ref: String,
-  val type: NotificationType,
-  val message: Notification
-)
+abstract class NotificationEvent{
+  abstract val scope: NotificationScope
+  open val ref: String? = null
+  abstract val type: NotificationType
+  open val message: Notification? = null
+}
 
+data class UnhealthyNotificationEvent(
```

Comment:
@emjburns calling for your attention here -- I've created a new data class specific to the `UnhealthyNotificationEvent`. I might change the implementation in the `Notifier` class when I'll remove the echo dependencies later on.